ministryofjustice/laa-apply-for-criminal-legal-aid
# CRIMAPP-893 Skip Means Assessment for Crown Court Appeals Without Changes on Apply

## Description of change

Skip Means Assessment for Crown Court Appeals Without Changes on Apply
Changes the appeal without changes logic so that means assessment is skipped entirely.
Ensures the `appeal_without_changes?` method handles when an appeal is updated from without changes to with changes.
Moves `#appeal_no_changes` from the `MeansPassporter` to `requires_means_assessment`.
Improves the completeness validation of employment questions.

## Link to relevant ticket

https://dsdmoj.atlassian.net/browse/CRIMAPP-893

## Notes for reviewer

This PR introduces a logic change to Apply.

When a case is appealed to Crown Court without changes and an application was submitted for the original case, the caseworker manually copies over the means details in MAAT from the previous application to the Appeal record.

An appeal without changes application is not necessarily Means Passported. It has the same means as the original application.

## Screenshots of changes (if applicable)

### Before changes:

### After changes:

## How to manually test the feature

1, Confirm that an Appeal to Crown court application with changes can be updated to an Appeal to Crown court without changes
2, Confirm that the Employment question is not asked.
3, Confirm that the application can be returned and resubmitted.
4, On a means tested application, confirm that the task lists updates correctly when the employment information is changed.
